src/docfx/lib/log/Telemetry.cs

```diff
@@ -233,7 +234,8 @@ public static void TrackException(Exception ex)
 
         public static void Flush()
         {
-            s_telemetryClient.Flush();
+            // TelemetryClient.Flush may meet deadlocks: https://github.com/microsoft/ApplicationInsights-dotnet/issues/1186
+            Task.WaitAny(Task.Run(s_telemetryClient.Flush), Task.Delay(2000));
```

Comment:
We can increase the timeout a little bit if 2 seconds is to small to flush all the telemetry.

Reply:
like 10 seconds?

Reply:
Sounds good